docs/APIs/storage/block_device.md

```diff
@@ -0,0 +1,38 @@
+# Block Devices
+
+The block device API provides a simple interface for providing access to block-based storage. A block device can be used to back a full [filesystem](filesystem.md) or can be written to
+directly.
+
+The full C++ API can be found [here](https://github.com/ARMmbed/mbed-os/blob/master/features/filesystem/bd/BlockDevice.h)
```

Comment:
I wasn't exactly sure on the syntax for linking to examples. Also this is in a header file, should it be duplicated to a repo on developer.mbed.org?